filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilderFactory.scala

```diff
@@ -1,50 +1,115 @@
 package cromwell.filesystems.blob
 
 import akka.actor.ActorSystem
+import com.azure.core.credential.AzureSasCredential
 import com.azure.core.management.AzureEnvironment
 import com.azure.core.management.profile.AzureProfile
 import com.azure.identity.DefaultAzureCredentialBuilder
 import com.azure.resourcemanager.AzureResourceManager
-import com.azure.storage.blob.BlobContainerClientBuilder
+import com.azure.resourcemanager.storage.models.{StorageAccount, StorageAccountKey}
+import com.azure.storage.blob.nio.AzureFileSystem
 import com.azure.storage.blob.sas.{BlobContainerSasPermission, BlobServiceSasSignatureValues}
+import com.azure.storage.blob.{BlobContainerClient, BlobContainerClientBuilder}
 import com.azure.storage.common.StorageSharedKeyCredential
 import com.typesafe.config.Config
 import cromwell.core.WorkflowOptions
 import cromwell.core.path.PathBuilderFactory
 import net.ceedubs.ficus.Ficus._
 
-import java.time.OffsetDateTime
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future
+import java.net.URI
+import java.nio.file.{FileSystem, FileSystemNotFoundException, FileSystems}
+import java.time.temporal.ChronoUnit
+import java.time.{Duration, Instant, OffsetDateTime}
+import scala.concurrent.{ExecutionContext, Future}
 import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
 
 final case class BlobFileSystemConfig(config: Config)
 final case class BlobPathBuilderFactory(globalConfig: Config, instanceConfig: Config, singletonConfig: BlobFileSystemConfig) extends PathBuilderFactory {
-  val container: String = instanceConfig.as[String]("store")
-  val endpoint: String = instanceConfig.as[String]("endpoint")
-  val workspaceId: Option[String] = instanceConfig.as[Option[String]]("workspace-id")
-  val workspaceManagerURL: Option[String] = singletonConfig.config.as[Option[String]]("workspace-manager-url")
+  val container: BlobContainerName = BlobContainerName(instanceConfig.as[String]("container"))
+  val endpoint: EndpointURL = EndpointURL(instanceConfig.as[String]("endpoint"))
+  val workspaceId: Option[WorkspaceId] = instanceConfig.as[Option[String]]("workspace-id").map(WorkspaceId(_))
+  val expiryBufferMinutes: Long = instanceConfig.as[Option[Long]]("expiry-buffer-minutes").getOrElse(10)
```

Comment:
Thinking out loud, once we have this base config checked into `reference.conf` it would be preferable to set the default there... but we can't do that yet, so here is good.